File: WEEK_4_C/Q6.c
#include <stdio.h>
#include <stdlib.h>

struct Node{ int data; struct Node* next; };

int main(){
    struct Node *head=NULL,*last=NULL,*t;
    for(int i=1;i<=4;i++){
        t=(struct Node*)malloc(sizeof(struct Node));
        t->data=i; t->next=head;
        if(!head){ head=t; head->next=head; last=head; }
        else{ last->next=t; t->next=head; last=t; }
    }
    t=head;
    do{ printf("%d ",t->data); t=t->next; }while(t!=head);
    return 0;
}

File: WEEK_3_Python/Q6.py
class Node:
    def __init__(self, data):
        self.data, self.next = data, None

def detect_loop(head):
    slow = fast = head
    while fast and fast.next:
        slow = slow.next
        fast = fast.next.next
        if slow == fast: return True
    return False

# Example
head = Node(1); head.next = Node(2); head.next.next = Node(3)
head.next.next.next = head  
print("Loop detected" if detect_loop(head) else "No loop")
